src/protocolstatus.cpp

```diff
@@ -43,11 +43,10 @@ enum RequestedInfo_t : uint16_t {
 
 void ProtocolStatus::onRecvFirstMessage(NetworkMessage& msg)
 {
-	uint32_t ip = getIP();
-	if (ip != 0x0100007F) {
-		std::string ipStr = convertIPToString(ip);
-		if (ipStr != g_config.getString(ConfigManager::IP)) {
-			std::map<uint32_t, int64_t>::const_iterator it = ipConnectMap.find(ip);
+	Connection::Address ip = getIP();
+	if (!ip.is_loopback()) {
+		if (ip != Connection::Address::from_string(g_config.getString(ConfigManager::IP))) {
```

Comment:
Digging through the implementation of asio is quite hard, but it's non trivial, results in a syscall so I'll cache it in the Connection.